### PR TITLE
sqlline: init at 1.12

### PR DESCRIPTION
**Pull request by taranarmo**
CLI for various DBMS via JDMC
https://github.com/julianhyde/sqlline
https://github.com/julianhyde/sqlline/releases/tag/sqlline-1.12.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/sq/sqlline/package.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      --add-flags "-jar $out/share/java/sqlline-${version}.jar"
+  '';
+
+  meta = with lib; {
+    description = "Shell for issuing SQL to relational databases via JDBC";
+    homepage = "https://github.com/julianhyde/sqlline";
+    license = licenses.bsd3;
+    platforms = platforms.all;
+    maintainers = with maintainers; [ taranarmo ];
+  };
```
Comment (Aleksanaa) [inline pkgs/by-name/sq/sqlline/package.nix:39]:
```suggestion
  meta = {
    description = "Shell for issuing SQL to relational databases via JDBC";
    homepage = "https://github.com/julianhyde/sqlline";
    license = lib.licenses.bsd3;
    platforms = lib.platforms.all;
    maintainers = with lib.maintainers; [ taranarmo ];
  };
```

**Review by Aleksanaa — COMMENTED** [review]
Also your commit message, it's "sqlline" not "sqllite"

**Comment by taranarmo** [comment]
@Aleksanaa, thanks a lot, I mistyped sqlline as sqllite several times, and missed this one 😅 
Also I updated the package.nix according to your suggestions.

**Comment by taranarmo** [comment]
Dear @Aleksanaa, sorry for being annoying, it's my first attempt to build anything with Maven and Nix. Initially I started it on `nix-darwin` which was able to build the package but rebuild on Linux failed. I had to patch doc generation of the package to be able to work offline. Now the building finishes on both platforms. Could you please take a look another time?

File: pkgs/by-name/sq/sqlline/package.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    sed -i 's|PUBLIC "-//OASIS//DTD DocBook XML V4.1.2//EN"|PUBLIC "-//OASIS//DTD DocBook XML V4.2//EN"|' src/docbkx/manual.xml
+  '';
+
+  buildOffline = true;
+
+  installPhase = ''
+    mkdir -p $out/share/java $out/bin
+    cp target/sqlline-${version}-jar-with-dependencies.jar $out/share/java/sqlline-${version}.jar
+    makeWrapper ${jre}/bin/java $out/bin/sqlline \
+      --add-flags "-jar $out/share/java/sqlline-${version}.jar"
```
Comment (Aleksanaa) [inline pkgs/by-name/sq/sqlline/package.nix:44]:
```suggestion
    runHook preInstall
    mkdir -p $out/share/java $out/bin
    cp target/sqlline-${version}-jar-with-dependencies.jar $out/share/java/sqlline-${version}.jar
    makeWrapper ${jre}/bin/java $out/bin/sqlline \
      --add-flags "-jar $out/share/java/sqlline-${version}.jar"
    runHook postInstall
```

File: pkgs/by-name/sq/sqlline/package.nix [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    docbook_xsl
+  ];
+
+  mvnParameters = "-DskipTests";
+
+  # Patch the DOCTYPE declaration in manual.xml
+  patchPhase = ''
+    sed -i 's|https://docbook.org/xml/4.2/docbookx.dtd|${docbook_xml_dtd_42}/xml/dtd/docbook/docbookx.dtd|' src/docbkx/manual.xml
+    sed -i 's|PUBLIC "-//OASIS//DTD DocBook XML V4.1.2//EN"|PUBLIC "-//OASIS//DTD DocBook XML V4.2//EN"|' src/docbkx/manual.xml
+  '';
```
Comment (Aleksanaa) [inline pkgs/by-name/sq/sqlline/package.nix:36]:
```suggestion
  postPatch = ''
    sed -i 's|https://docbook.org/xml/4.2/docbookx.dtd|${docbook_xml_dtd_42}/xml/dtd/docbook/docbookx.dtd|' src/docbkx/manual.xml
    sed -i 's|PUBLIC "-//OASIS//DTD DocBook XML V4.1.2//EN"|PUBLIC "-//OASIS//DTD DocBook XML V4.2//EN"|' src/docbkx/manual.xml
  '';
```

File: pkgs/by-name/sq/sqlline/package.nix [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    makeWrapper ${jre}/bin/java $out/bin/sqlline \
+      --add-flags "-jar $out/share/java/sqlline-${version}.jar"
+  '';
+
+  meta = {
+    description = "Shell for issuing SQL to relational databases via JDBC";
+    homepage = "https://github.com/julianhyde/sqlline";
+    license = lib.licenses.bsd3;
+    platforms = lib.platforms.all;
+    maintainers = with lib.maintainers; [ taranarmo ];
```
Comment (Aleksanaa) [inline pkgs/by-name/sq/sqlline/package.nix:52]: missing `mainProgram`

File: pkgs/by-name/sq/sqlline/package.nix [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    repo = "sqlline";
+    rev = "sqlline-${version}";
+    hash = "sha256-rUlGtMgTfhciQVif0KaUcuY28wh+PrHsKen8qODom24=";
+  };
+
+  mvnHash = "sha256-Lalsseo6kwvaSXgXllE6y05qoeS+aqLx1N8JT4sQwqU=";
+
+  nativeBuildInputs = [
+    makeWrapper
+    docbook_xml_dtd_42
```
Comment (Aleksanaa) [inline pkgs/by-name/sq/sqlline/package.nix:26]: This may not be needed. Check yourself.

Comment (taranarmo) [reply pkgs/by-name/sq/sqlline/package.nix:25]: Removed

**Comment by taranarmo** [comment]
Thanks for the suggestions and comments. Basically I applied them all (and made some typos and fixed them).

**Comment by Aleksanaa** [comment]
@ofborg build sqlline

**Comment by taranarmo** [comment]
@ofborg build sqlline

File: pkgs/by-name/sq/sqlline/package.nix [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+}:
+
+maven.buildMavenPackage rec {
+  pname = "sqlline";
+  version = "1.12.0";
+
+  src = fetchFromGitHub {
+    owner = "julianhyde";
+    repo = "sqlline";
+    rev = "sqlline-${version}";
```
Comment (pluiedev) [inline pkgs/by-name/sq/sqlline/package.nix:17]: See https://github.com/NixOS/nixpkgs/pull/355973 (you might need to rebase to latest master) 
```suggestion
    tag = "sqlline-${version}";
```

File: pkgs/by-name/sq/sqlline/package.nix [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    makeWrapper
+    docbook_xml_dtd_42
+  ];
+
+  mvnParameters = "-DskipTests";
+
+  # Patch the DOCTYPE declaration in manual.xml
+  postPatch = ''
+    sed -i 's|https://docbook.org/xml/4.2/docbookx.dtd|${docbook_xml_dtd_42}/xml/dtd/docbook/docbookx.dtd|' src/docbkx/manual.xml
+    sed -i 's|PUBLIC "-//OASIS//DTD DocBook XML V4.1.2//EN"|PUBLIC "-//OASIS//DTD DocBook XML V4.2//EN"|' src/docbkx/manual.xml
```
Comment (pluiedev) [inline pkgs/by-name/sq/sqlline/package.nix:33]: Don't `sed` when you can `substituteInPlace` (mostly for better error reporting)
```suggestion
    substituteInPlace src/docbkx/manual.xml \
      --replace-fail "https://docbook.org/xml/4.2/docbookx.dtd" "${docbook_xml_dtd_42}/xml/dtd/docbook/docbookx.dtd" \
      --replace-fail 'PUBLIC "-//OASIS//DTD DocBook XML V4.1.2//EN"' 'PUBLIC "-//OASIS//DTD DocBook XML V4.2//EN"'
```

File: pkgs/by-name/sq/sqlline/package.nix [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+  '';
+
+  buildOffline = true;
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/share/java $out/bin
+    cp target/sqlline-${version}-jar-with-dependencies.jar $out/share/java/sqlline-${version}.jar
```
Comment (pluiedev) [inline pkgs/by-name/sq/sqlline/package.nix:41]: Nit
```suggestion
    mkdir -p $out/bin
    install -D target/sqlline-${version}-jar-with-dependencies.jar $out/share/java/sqlline-${version}.jar
```

File: pkgs/by-name/sq/sqlline/package.nix [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+      --add-flags "-jar $out/share/java/sqlline-${version}.jar"
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "Shell for issuing SQL to relational databases via JDBC";
+    homepage = "https://github.com/julianhyde/sqlline";
+    mainProgram = "sqlline";
+    license = lib.licenses.bsd3;
+    platforms = lib.platforms.all;
```
Comment (pluiedev) [inline pkgs/by-name/sq/sqlline/package.nix:52]: Would probably better if it's all platforms that Java supports - then again, I wonder if  buildMavenPackage already sets this for you (?)
```suggestion
	inherit (jre.meta) platforms;
```

**Review by pluiedev — CHANGES_REQUESTED** [review]

File: pkgs/by-name/sq/sqlline/package.nix (hunk 9 above)
Comment (taranarmo) [reply pkgs/by-name/sq/sqlline/package.nix:52]: It does indeed, I will delete this tag completely and let `buildMavenPackage` handle it

**Comment by Aleksanaa** [comment]
@ofborg build sqlline

**Pull request merged**